File: Kata 7/Tartamudo/Tartamudo.py
"""
Kata Level: 7.

Examples:

accum("abcd") -> "A-Bb-Ccc-Dddd"
accum("RqaEzty") -> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
accum("cwAt") -> "C-Ww-Aaa-Tttt"

The parameter of accum is a string which includes only letters from a..z and A..Z.
"""

# Mi solución en Python.

def accum(s):
    result = ""
    for i, letter in enumerate(s.lower()):
        result += letter.upper()
        while i > 0:
            result += letter
            i -= 1
        result += "-"
    return result[:-1]

# Solución de la comunidad 1.

def accum_community(s):
    return '-'.join(c.upper() + c.lower() * i for i, c in enumerate(s))

print(accum("abcd"))
print(accum("RqaEzty"))

File: Kata 7/Tartamudo/Tartamudo.cs
/* Kata Level: 7.
 * 
 * Examples:
 * 
 * accum("abcd") -> "A-Bb-Ccc-Dddd"
 * accum("RqaEzty") -> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
 * accum("cwAt") -> "C-Ww-Aaa-Tttt"
 * 
 * The parameter of accum is a string which includes only letters from a..z and A..Z.
 */

using System;

public class Accumul 
{
	public static String Accum(string s) 
    {
        string result = "";
        for(int i = 0 ; i < s.Length ; i++) 
        {
            result +=  s[i].ToString().ToUpper();
            for(int j = 0 ; j < i ; j++)   result +=   s[i].ToString().ToLower();
            result += "-";
        }
        return result.Substring(0, result.Length - 1);
    }
}